libs/language-server/src/lib/ast/expressions/internal-value-representation.ts

```diff
@@ -14,7 +19,50 @@ import {
   isTransformDefinition,
   isValuetypeAssignment,
 } from '../generated/ast';
-import type { WrapperFactoryProvider } from '../wrappers';
+import { type WrapperFactoryProvider } from '../wrappers';
+
+abstract class JayveeError extends Error {
+  abstract override name: string;
+
+  constructor(message: string) {
+    super(message);
+  }
+
+  override toString(): string {
+    return `${this.name}: ${this.message}`;
+  }
+
+  abstract clone(): JayveeError;
+}
+
+export class InvalidError extends JayveeError {
```

Comment:
I think we should call it `InvalidValue` instead. But might be personal preference, opinions?